src/psyclone/tests/domain/lfric/lfric_field_test.py

```diff
@@ -197,9 +198,12 @@ def test_lfricfields_call_err():
     fld_arg._intrinsic_type = "triple-type"
     with pytest.raises(InternalError) as err:
         LFRicFields(invoke)._invoke_declarations(ModuleGen(name="my_mod"))
+    test_str = str(err.value)
+    if six.PY2:
```

Comment:
For safety it might be worth adding 1 as an argument to str.replace so you only replace the first occurrence, otherwise you might accidentally change e.g. `u'you'` to `'yo'`. I would suggest that you do this for all the cases where you add in six.PY2 to remove the u

Reply:
Another possible option to try that would not need PY2 would be to add u to the start of the assert string as Python3 is happy with that and Python2 will then create a unicode string.

Reply:
> For safety it might be worth adding 1 as an argument to str.replace so you only replace the first occurrence, otherwise you might accidentally change e.g. `u'you'` to `'yo'`. I would suggest that you do this for all the cases where you add in six.PY2 to remove the u

In reply to myself, this obviously won't work if the output has multiple `u'xxx'` cases in the string e.g. from a list. So please ignore in these cases.

Reply:
>
> Hmmm, having said that I have just read that in Python 3 from 3.0-3.2 adding u before a string gives an error. Perhaps stick with the PY2 solution then.

I think I'll stick to PY2 solution for now as it will be easier to find and remove when we drop the support for Pyhon 2.